src/app/clusters/ethernet-network-diagnostics-server/ethernet-diagnostics-cluster.h

```diff
@@ -0,0 +1,74 @@
+/*
+ *    Copyright (c) 2025 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+
+#include <app/server-cluster/DefaultServerCluster.h>
+#include <app/server-cluster/OptionalAttributeSet.h>
+#include <clusters/EthernetNetworkDiagnostics/ClusterId.h>
+#include <clusters/EthernetNetworkDiagnostics/Enums.h>
+#include <clusters/EthernetNetworkDiagnostics/Metadata.h>
+#include <lib/core/DataModelTypes.h>
+#include <platform/DiagnosticDataProvider.h>
+#include <protocols/interaction_model/StatusCode.h>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+
+struct EthernetDiagnosticsEnabledAttributes
+{
+    bool enableCarrierDetect : 1;
+    bool enableFullDuplex : 1;
+    bool enablePHYRate : 1;
+    bool enableTimeSinceReset : 1;
+};
+
+class EthernetDiagnosticsServerCluster : public DefaultServerCluster
+{
+public:
+    using OptionalAttributeSet = chip::app::OptionalAttributeSet<
+        EthernetNetworkDiagnostics::Attributes::CarrierDetect::Id, EthernetNetworkDiagnostics::Attributes::FullDuplex::Id,
+        EthernetNetworkDiagnostics::Attributes::PHYRate::Id, EthernetNetworkDiagnostics::Attributes::TimeSinceReset::Id,
```

Comment:
I think we should remove from here the packet and error counts because those are controlled by the feature map. There are only 4 real optional attributes.